src/promptflow-devkit/promptflow/_sdk/entities/_flows/flex.py

```diff
@@ -33,9 +34,14 @@ def __init__(
         code = Path(code)
         # entry function name
         self.entry = entry
+        self._flow_dir, self._flex_file_name = resolve_flow_path(path)
         # TODO(2910062): support non-dag flow execution cache
         super().__init__(code=code, path=path, dag=data, content_hash=None, **kwargs)
 
+    @property
+    def name(self) -> str:
+        return self._flow_dir.name
```

Comment:
FlexFlow can have name in yaml? return just the flow_dir name might not have bug

Reply:
Synchronized offline, no modifications.